### DISCUSSION TITLE
Close Tab Without Right Clicking or Expanding

### DISCUSSION THREAD

**Discussion by sirinath** [zen-browser/desktop#1083] [Ideas]
Can we get a close option in the tool tip

![image](https://github.com/user-attachments/assets/bec1e16b-b015-4d99-8962-464b4788a41e)

Like we have here:

![image](https://github.com/user-attachments/assets/3f91e975-347d-40b3-af63-6cfbc6ec43e5)

To close the tab.

**Comment by kolemcrae** [comment]
Its the main thing stopping me from using the condensed view - I often close tabs and the extra click is super annoying. It can be in the tooltip or maybe a way that on hover it expands just that tab so I can close it - not a fan of the whole sidebar expanding on hover.

**Reply by mr-cheffy** [reply]
You can double click, right click or ctrl+W

**Reply by kolemcrae** [reply]
I guess what we are saying is we would like a single click option - double click also doesnt seem to do anything for me, other than open the tab. So right now its either right click, then select the option or a memorized keyboard shortcut.

**Reply by kolemcrae** [reply]
Oh wait, you meant middle click didn't you? I guess thats an option, but I've not used that in the past so its a new muscle memory to learn rather than the normal click the X behavior.

**Reply by AlTimofeyev** [reply]
I was just thinking about this too. The **ctrl+W** shortcut is super useful for closing the current tab quickly. But if I need to close multiple tabs that I'm not currently viewing, I was thinking of using a **middle click**, or a **ctrl+click** (**cmd+click** on mac), to close multiple tabs without viewing them. I tried adding this in the shortcuts settings but a mouse/trackpad click won't register in the shortcuts unfortunately. Maybe it could be one of the built-in shortcuts by default later on?

**Reply by mr-cheffy** [reply]
There's a theme that I'm going to merge that displays the close button when collapsed

**Reply by AlTimofeyev** [reply]
Thank you!! Gonna test it out when it's released.

**Reply by sirinath** [reply]
You can do something like this:

![image](https://github.com/user-attachments/assets/8e9479b3-b60c-4d3f-b0ff-99e423515164)

It will need nicer and perhaps smaller and streamlined icons than what I have show above,

Maybe you can have a menu bar appearing next to the preview panel giving more option depending if it is active or not.

**Reply by DavidSint** [reply]
Perhaps something like this where the x icon appears on hover?

![image](https://github.com/user-attachments/assets/1503c58c-527e-46ab-9a91-3514d85a8c52)

(forgive my poor designing)

**Reply by Nitrousoxide** [reply]
> Perhaps something like this where the x icon appears on hover?
> 
> ![image](https://private-user-images.githubusercontent.com/44842045/370470865-1503c58c-527e-46ab-9a91-3514d85a8c52.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcyNzM2ODQsIm5iZiI6MTcyNzI3MzM4NCwicGF0aCI6Ii80NDg0MjA0NS8zNzA0NzA4NjUtMTUwM2M1OGMtNTI3ZS00NmFiLTlhOTEtMzUxNGQ4NWE4YzUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTI1VDE0MDk0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBiMTdjNWJjMDMzMDI5MzE0NjM5ZTYzN2I0MmUxN2JkY2Y0MDdjZmJhYmUwNGY0MTA4YTg4MDY3OWJhYTFlNDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZgNWKTDFhuurjUMUesMs6NkJUdx65Ua8HiP92xXuPjo)
> 
> (forgive my poor designing)

The worry I have with something like that is that it's very easy to accidentally close the tab.

**Reply by AlTimofeyev** [reply]
I like the approach @DavidSint presented, and taking @Nitrousoxide concern into account, it would be nice if on the CSS side of things the content could identify when it is hovered and a specific key is pressed in order to change the click function into a close function. 

Example:  
- Hovering over a tab and clicking it allows you to switch to that tab.  
- Holding down the **`CTRL`** / **`CMD`** key and then hovering over a tab and clicking it allows you to close that tab.

**Reply by MannixMD** [reply]
> Perhaps something like this where the x icon appears on hover?
> 
> ![image](https://private-user-images.githubusercontent.com/44842045/370470865-1503c58c-527e-46ab-9a91-3514d85a8c52.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2ODcxMDIsIm5iZiI6MTcyNzY4NjgwMiwicGF0aCI6Ii80NDg0MjA0NS8zNzA0NzA4NjUtMTUwM2M1OGMtNTI3ZS00NmFiLTlhOTEtMzUxNGQ4NWE4YzUyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTMwVDA5MDAwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwNWE5ZWVkZGY3YmIzOGI5YmRiNjMzMzM5NmExNmIwZGNhM2QwZjU2MzA3MmYyNGMxZGMyMmJkMWE2MjJjMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.HQpIOgakddV2lCXrt2OAin6l-EN1b0J470OHBgQdi-Q)
> 
> (forgive my poor designing)

There is an [addon](https://zen-browser.app/themes/49dbaa98-06ee-42bd-9a8e-834babef7a41) in Zen store that adds the close button on hover :)

**Comment by thyeun** [comment]
middle-click can close it, pls dont add the ```x``` to close it, it is ugly.

**Comment by Manut38** [comment]
I very much agree with @thyeun, I really think you should start using the middle mouse button for closing tabs. It's how I've been using my browsers for the last 8 years or so and it makes everything sooo much quicker!

If you're planning to add this 'close' button to the compact tabs, I would prefer to be able to disable it in the settings.

**Comment by Nitrousoxide** [comment]
Laptops rarely have a middle mouse button.

**Reply by Manut38** [reply]
They kind of do! You can set up the touchpad to click the middle mouse button when you touch it with three fingers. I use that in Windows and it works the same for me in all Linux distros i've tried.

**Reply by anatasiajp** [reply]
> Laptops rarely have a middle mouse button.

MouseGesture is the savior, right @mauro-balades ?

**Reply by thyeun** [reply]
who said laptop dont have middle click button, all laptop have even 12 years old laptop have that, so why not on newer laptop, you just dont know how to using your laptop.

**Comment by qudo-code** [comment]
can we add a little "x" to the tab boxes (like chrome and firefox do)? its really annoying to have to right click. Or how about cmd/ctrl click closes them? or, how about there's a like "edit tabs" button that makes all the tab squares jiggle like iOS and you can go through and mass delete specific ones easy. Also please don't solve this with another mouse button (middle click), many of us don't like using a mouse when we can.
![image](https://github.com/user-attachments/assets/0687d514-7539-4304-85b7-53317864bc07)

**Reply by thyeun** [reply]
Why you need right click? you can using middle click to close the tab and open new tab, why need to add little ```x``` to close it, it is ugly with that, some more if you on youtube, you will know how trouble to have that. If you prefer using keyboard, you can press ```Ctrl + w``` to close it, if you have the ```x``` close it, you still using your mouse to click it, you already request using mouse the click the ```x``` why not using middle click?

**Reply by tedbergstrand** [reply]
burn0rice made a Zen Mod to do this: https://zen-browser.app/themes/49dbaa98-06ee-42bd-9a8e-834babef7a41.

**Comment by universish** [comment]
https://github.com/zen-browser/desktop/discussions/2866

> 2. I really want the “Close this open tab” feature, because when the “Expand Sidebar” button (the icon with the top arrow to the right and the bottom arrow to the left) is closed, the logos of the open tabs appear. Their names are not written. The “close tab” icon in the shape of an “x” does not appear either. I have to close the tab by hovering over it > Right click > Click close tab, which is a pain when I want to close many tabs.
> 
> When the “Expand Sidebar” button is on, the logos and names of the tabs appear, and the “Close tab” icon (x) appears when you mouse over it. But when the “Expand Sidebar” button is on, it takes up a lot of space on my screen, even though I have shortened it, it takes up a lot of space. the reduction of screen space makes it difficult to use on a laptop.
> 
> So it would be very useful if a “Close this open tab” button could be created or improved (there is no such button in zen browser yet, only “Close tab”). If a “Close tab” button is made, there may be errors in the system. It may not know which tab to close. That's why I suggest creating a “Close this open tab” button.
